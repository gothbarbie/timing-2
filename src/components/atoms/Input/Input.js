import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const InputStyle = styled.input`
  border-radius: 4px;
  border: 1px solid ${({ theme }) => theme.palette.grayScale[0]};
  padding: 1rem 1.5rem;
  font-size: 1.6rem;
  color: ${({ theme }) => theme.palette.grayScale[0]};
  outline: none;

  &:focus {
    border: 1px solid ${({ theme }) => theme.palette.primary[0]};
  }
`

const Input = ({ name, onChange, type = 'text' }) => {
  return <InputStyle name={name} onChange={onChange} type={type} />
}

Input.propTypes = {
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  type: PropTypes.string,
}

export default Input
