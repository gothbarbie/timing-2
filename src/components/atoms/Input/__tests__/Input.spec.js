import React from 'react'
import { shallow } from 'enzyme'

import Input from '../Input'

describe('components/atoms/Input', () => {
  it('renders', () => {
    const component = shallow(<Input />)

    expect(component).toMatchSnapshot()
  })
})
