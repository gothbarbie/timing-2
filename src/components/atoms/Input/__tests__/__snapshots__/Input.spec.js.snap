// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/Input renders 1`] = `
<InputStyle
  type="text"
/>
`;
